pilot/pkg/model/sidecar.go

```diff
@@ -980,7 +981,7 @@ func (sc *SidecarScope) appendSidecarServices(servicesAdded map[host.Name]sideca
 		// Update index as well, so that future reads will merge into the new service
 		foundSvc.svc = copied
 		servicesAdded[foundSvc.svc.Hostname] = foundSvc
-		sc.servicesByHostname[s.Hostname] = s
+		sc.servicesByHostname[s.Hostname] = copied
```

Comment:
This seems really subtle; should we add a comment?

Reply:
Added a test to gurantee. Yes, actually `copied` is the merged Service